Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -0,0 +1,228 @@
+/-
+Copyright (c) 2023 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import Mathlib.MeasureTheory.Measure.ProbabilityMeasure
+
+/-!
+# The Lévy-Prokhorov distance on spaces of finite measures and probability measures
+
+## Main definitions
+
+* `MeasureTheory.levyProkhorovEDist`: The Lévy-Prokhorov distance between two measures.
+
+## Main results
+
+* `levyProkhorovEDist.pseudoEMetricSpace_finiteMeasure`: The Lévy-Prokhorov distance is a
+  pseudoemetric on the space of finite measures.
+* `levyProkhorovEDist.pseudoEMetricSpace_probabilityMeasure`: The Lévy-Prokhorov distance is a
+  pseudoemetric on the space of probability measures.
+
+## Todo
+
+* Show that in Borel spaces, the Lévy-Prokhorov distance is a metric; not just a pseudometric.
+* Prove that if `X` is metrizable and separable, then the Lévy-Prokhorov distance metrizes the
+  topology of weak convergence.
+
+## Tags
+
+finite measure, probability measure, weak convergence, convergence in distribution, metrizability
+-/
+
+open Topology Metric Filter Set ENNReal NNReal
+
+namespace MeasureTheory
+
+open scoped Topology ENNReal NNReal BoundedContinuousFunction
+
+section Levy_Prokhorov
+
+/-! ### Lévy-Prokhorov metric -/
+
+variable {ι : Type*} {Ω : Type*} [MeasurableSpace Ω]
+variable [MetricSpace Ω] [OpensMeasurableSpace Ω]
+
+/-- The Lévy-Prokhorov distance on the space of finite measures:
+`d(μ,ν) = inf {r ≥ 0 | ∀ B, μ B ≤ ν Bᵣ + r ∧ ν B ≤ μ Bᵣ + r}`. -/
+noncomputable def levyProkhorovEDist (μ ν : Measure Ω) : ℝ≥0∞ :=
+  sInf {ε | ∀ B, MeasurableSet B →
+            μ B ≤ ν (thickening ε.toReal B) + ε ∧ ν B ≤ μ (thickening ε.toReal B) + ε}
+
+lemma measure_apply_le_of_le_of_forall_le_measure_thickening_add {ε₁ ε₂ : ℝ≥0∞} (μ ν : Measure Ω)
+    (h_le : ε₁ ≤ ε₂) {B : Set Ω} (hε₁ : μ B ≤ ν (thickening ε₁.toReal B) + ε₁):
+    μ B ≤ ν (thickening ε₂.toReal B) + ε₂ := by
+  by_cases ε_top : ε₂ = ∞
+  · simp only [ne_eq, FiniteMeasure.ennreal_coeFn_eq_coeFn_toMeasure, ε_top, top_toReal,
+                add_top, le_top]
+  apply hε₁.trans (add_le_add ?_ h_le)
+  exact measure_mono (μ := ν) (thickening_mono (toReal_mono ε_top h_le) B)
```

Comment:
This belongs to an earlier file, right?

Reply:
This feels quite Lévy-Prokhorov-specific to me, but I have nothing else against moving it.

(One way to see that it is rather specific is that it involves a sum of a measure and a distance: two things that would normally have totally different units and should not be added together. I can only think of Lévy-Prokhorov metric and some coupling arguments which eventually argue with Lévy-Prokhorov metric, which would "mix quantities of different units" in this way.)

Reply:
I don't have strong opinions, but it might be good to record whatever your decision is in a comment or docstring.